lib/Basic/Statistic.cpp

```diff
@@ -0,0 +1,297 @@
+//===--- Statistic.cpp - Swift unified stats reporting --------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Basic/Statistic.h"
+#include "swift/Driver/DependencyGraph.h"
+#include "swift/SIL/SILModule.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/Path.h"
+#include "llvm/Support/Process.h"
+#include "llvm/Support/raw_ostream.h"
+#include <chrono>
+
+namespace swift {
+using namespace llvm;
+using namespace llvm::sys;
+
+static std::string
+makeFileName(StringRef ProcessName) {
+  std::string tmp;
+  raw_string_ostream stream(tmp);
+  auto now = std::chrono::system_clock::now();
+  stream << "stats"
+         << "-" << now.time_since_epoch().count()
+         << "-" << ProcessName
+         << "-" << Process::GetRandomNumber()
```

Comment:
::getpid is unistd.h, not on windows. There's a a _getpid in process.h, but then I'd be adding these things to LLVM's Process.inc files and I wanted to keep the patch as small as possible.